library/core/src/iter/traits/iterator.rs

```diff
@@ -3011,6 +3015,18 @@ pub trait Iterator {
     /// assert_eq!(v_cloned, vec![1, 2, 3]);
     /// assert_eq!(v_map, vec![1, 2, 3]);
     /// ```
+    ///
+    /// To get the best performance, try to clone late:
+    ///
+    /// ```
+    /// let a = [vec![0_u8, 1, 2], vec![3, 4], vec![23]];
+    /// // don't do this:
+    /// let bad: Vec<_> = a.iter().cloned().filter(|s| s.len() == 1).collect();
+    /// assert_eq!(&[vec![23]], &bad[..]);
```

Comment:
I'd prefer to avoid value judgement terms in our docs. How about we rename these from `bad`/`good` to `slower`/`faster` or something similar?

Reply:
I like it! Done.